### PR TITLE
test: Retry auth in checkClientCertAuthentication

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
I did see this locally with 50% chance until I seriously started looking for reasons, then it stopped failing, of course. Current theory: it is something intermediate that goes away with retrying.

**Comment by mvollmer** [comment]
Experiment results: The first password auth attempt in checkClientCertAuthentication sometimes fails with an internal error. Retrying it after five seconds always succeeds. Delaying the first try by ten seconds also always succeeds.

I have no idea why it fails the first time and why it succeeds the second time. I don't think we need to find out.

**Comment by martinpitt** [comment]
So perhaps the `while ! id alice; do sleep 5; done` check isn't sufficient, and we need to wait until password auth works? That might also explain some related flakes.

**Comment by mvollmer** [comment]
> So perhaps the `while ! id alice; do sleep 5; done` check isn't sufficient, and we need to wait until password auth works?

During my testing, `id alice` either worked on first try, or would never succeed. The latter is probably https://github.com/samba-in-kubernetes/samba-container/issues/160.

How would you wait until passwd auth works? I think just retrying inside do_test is ok.

**Comment by martinpitt** [comment]
@mvollmer Note that https://github.com/samba-in-kubernetes/samba-container/issues/160 applied to the other container image that we used in the meantime. I went back to https://github.com/Fmstrat/samba-domain, and it at least made that problem better. You can check `sssctl domain-status cockpit.lan` if it's the same problem, but that was very binary: It either worked right away or never. So if you can "fix" this by retrying, it's definitively something else.

Not sure about checking if pwd auth works -- in that case we can just retry the whole curl thing I suppose. There's only so much yak that we can and want to shave :grin:

**Comment by mvollmer** [comment]
Slightly amplified test run: https://cockpit-logs.us-east-1.linodeobjects.com/pull-19719-20231211-084650-1638835f-rhel-9-4-networking/log.html

File: test/verify/check-system-realms [hunk 1]
```diff
@@ -397,11 +397,16 @@ class CommonTests:
...
+
+            def try_auth():
+                output = m.execute(['curl', '-ksS', '-D-', *authopts, 'https://localhost:9090/cockpit/login'])
+                for s in expected:
+                    self.assertIn(s, output)
+                for s in (not_expected or []):
+                    self.assertNotIn(s, output)
+                return True
+
+            testlib.wait(try_auth, delay=5, tries=10)
```
Comment (martinpitt) [inline test/verify/check-system-realms:409]: Could the test do that retry the first time only? We want to treat that as "async init"/"warming up". We totally don't expect this to be wonky afterwards, that could hide bugs in our own code as well.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Argh, the faulty flake8 unit-test container update does not affect just bots https://github.com/cockpit-project/bots/issues/5659 but also cockpit. That is urgent I suppose.

Note that this is still no-test.

Thanks!

File: test/verify/check-system-realms (hunk 1 above)
Comment (mvollmer) [reply test/verify/check-system-realms:409]: Yes, indeed.

File: test/verify/check-system-realms [hunk 2]
```diff
@@ -427,11 +435,13 @@ class CommonTests:
 
         # from sssd
         self.allow_journal_messages("alice is not allowed to run sudo on x0.  This incident will be reported.")
+        # occasional intermediate error during password auth
+        self.allow_journal_messages("cockpit-session: user account access failed: 4 alice: System error")
```
Comment (martinpitt) [inline test/verify/check-system-realms:439]: The "4" is funny. But we can keep this for the time being, and replace with `.*` if/when it becomes a problem.

**Review by martinpitt — APPROVED** [review]
Danke! I bow to the bots in their infinite wisdom for the final verdict, of course. :bow:

**Comment by martinpitt** [comment]
Meh, [TestAD failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19719-20231211-123554-6dab14a5-fedora-39-firefox-networking/log.html#34). That's a different test, but possibly same root cause?

**Comment by martinpitt** [comment]
After 5 retries, I'll [look the other way](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19719-20231211-123554-6dab14a5-fedora-39-devel/log.html) here.

**Pull request merged**